src/main/java/org/truffleruby/language/arguments/RubyArguments.java

```diff
@@ -115,31 +118,37 @@
         return newArgs;
     }
 
-    public static boolean assertFrameArguments(Object[] frameArguments) {
-        Object callerData = getCallerData(frameArguments);
-        InternalMethod method = getMethod(frameArguments);
-        DeclarationContext declarationContext = getDeclarationContext(frameArguments);
-        Object self = getSelf(frameArguments);
-        Object block = getBlock(frameArguments);
-        Object[] arguments = getArguments(frameArguments);
-
-        assert method != null;
-        assert declarationContext != null;
-        assert self != null;
+    public static boolean assertArguments(Object[] arguments) {
         assert arguments != null;
-        assert ArrayUtils.assertValidElements(arguments, arguments.length);
-
-        assert callerData == null ||
-                callerData instanceof MaterializedFrame ||
-                callerData instanceof SpecialVariableStorage ||
-                callerData instanceof FrameAndVariables;
-
-        /* The block in the frame arguments is always either a Nil or RubyProc. The provision of Nil if the caller
-         * doesn't want to provide a block is done at the caller, because it will know the type of values within its
-         * compilation unit.
-         *
-         * When you read the block back out in the callee, you'll therefore get a Nil or RubyProc. */
-        assert block instanceof Nil || block instanceof RubyProc : block;
+        assert arguments.length >= RUNTIME_ARGUMENT_COUNT;
+
+        final Object declarationFrame = arguments[ArgumentIndicies.DECLARATION_FRAME.ordinal()];
+        assert declarationFrame == null || declarationFrame instanceof MaterializedFrame : declarationFrame.getClass();
```

Comment:
For the assert message, could you use `declarationFrame` (so no `.getClass()`) which is likely to show the class anyway but also maybe more useful output?
Same below

Reply:
For example `SpecialVariableStorage` was very confusing as a `toString`. I added this assert message to diagnose just that.

Reply:
We could add a `SpecialVariableStorage ` prefix in its toString.
For instance if the kwargs descriptor is in the wrong place, it'd be a useful help to also see which descriptor it is and not just it's class.
It's minor though (one can change while debugging if needed), so not a big deal, fine as-is.